libs/test-macros/src/lib.rs

```diff
@@ -51,11 +36,11 @@ pub fn test_connector(attr: TokenStream, input: TokenStream) -> TokenStream {
     };
 
     // Generate the final function
-    let include_tagged = &attrs.include_tagged;
-    let exclude_tagged = &attrs.exclude_tagged;
-    let capabilities = &attrs.capabilities;
-    let preview_features = &attrs.preview_features;
-    let namespaces = &attrs.namespaces;
+    let include_tagged = &attrs.include_tagged.0;
+    let exclude_tagged = &attrs.exclude_tagged.0;
+    let capabilities = &attrs.capabilities.0;
+    let preview_features = &attrs.preview_features.0;
+    let namespaces = &attrs.namespaces.0;
```

Comment:
Do we really need to have this `.0` at the end, or can we refactor `test_connector()` into something that resembles the originally returned struct more?

Reply:
We need the newtypes because wr can’t impl third-party traits for `Vec`s. Yes, it’s possible to get rid of them by introducing an intermediate data structure and mapping the data. It would make sense if this struct was exported and we wanted to hide the implementation details, but I don’t think there’s a good reason to do so for what already is a temporary intermediate structure private to one function.